components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Stack/Encoders/Models/EncodeableDictionary.cs

```diff
@@ -54,7 +70,23 @@ public class EncodeableDictionary : List<KeyDataValuePair>, IEncodeable {
 
         /// <inheritdoc/>
         public virtual void Decode(IDecoder decoder) {
-            // No operation. Cannot decode without known keys.
+            // Read keys for decoding. 
+            StringCollection keys = null;
+            try {
+                // May throw in some decoders if an empty array was encoded.
+                keys = decoder.ReadStringArray(kKeysIdentifier);
+            }
+            catch {
```

Comment:
Is this a good catch principle?

Reply:
No, it is not. But as I tried to put in the comments, because each decoder may throw a custom exception, we either have to enumerate all such exceptions or catch this broadly. For instance, the JsonDecoderEx will throw JsonReaderException, which the others obviously will not. If future decoders are added, these exceptions must either be standardized or put enumerated here. Thoughts @buzzfrog?

Reply:
you can't have a catch on a exception that are higher in the hierarchy?
What happens if we get a memory corrupt exception? (unlikely)

Reply:
can it even be a security issue? with buffer overrun?

Reply:
Good question. I don't think we can guarantee that all the different exceptions will derive from one in a higher chain. I think we will end up at the base. But do you think we should attempt to enumerate all the different exceptions instead?

Reply:
Can you not verify that it is not empty before you do decode?

Reply:
That is right. I cannot verify that. Unless we write a property for that, which we read first. Such as KeyLength. Thoughts on that approach?

Reply:
I have changed my approach to this. If we write the number of keys we don't have to deal with this. What do you think @buzzfrog?

Reply:
LGTM